src/main/java/com/microsoftopentechnologies/windowsazurestorage/service/UploadService.java

```diff
@@ -0,0 +1,168 @@
+/*
+ Copyright 2017 Microsoft Open Technologies, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package com.microsoftopentechnologies.windowsazurestorage.service;
+
+import com.microsoftopentechnologies.windowsazurestorage.AzureBlobMetadataPair;
+import com.microsoftopentechnologies.windowsazurestorage.Messages;
+import com.microsoftopentechnologies.windowsazurestorage.exceptions.WAStorageException;
+import com.microsoftopentechnologies.windowsazurestorage.helper.Constants;
+import com.microsoftopentechnologies.windowsazurestorage.service.model.PublisherServiceData;
+import com.microsoftopentechnologies.windowsazurestorage.service.model.UploadType;
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.Util;
+import org.apache.commons.lang.StringUtils;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.StringTokenizer;
+
+public abstract class UploadService extends StoragePluginService<PublisherServiceData> {
+    protected static final String zipFolderName = "artifactsArchive";
+    protected static final String zipName = "archive.zip";
+
+    protected UploadService(PublisherServiceData serviceData) {
```

Comment:
Change PublisherServiceData -> UploadServiceData, BuilderServiceData -> DownloadServiceData to make them consistent with service class name?